fvm/crypto/crypto.go

```diff
@@ -52,19 +66,35 @@ func (DefaultSignatureVerifier) Verify(
 	return valid, nil
 }
 
-// NewHasher returns a crypto hasher supported by runtime.
-func NewHasher(hashAlgo hash.HashingAlgorithm) hash.Hasher {
+func Hash(hashAlgo hash.HashingAlgorithm, tag string, data []byte) ([]byte, error) {
+	var hashFunc func(tag string, data []byte) hash.Hash
+
+	shaHashWrapper := func(f func([]byte) hash.Hash) func(tag string, data []byte) hash.Hash {
```

Comment:
This is totally functional and valid.

I have an idea to put more emphasis on the "tagged hashing" that I would like us to use across the repos (fvm and sdk for now) for signing and verifying, and I can start it here. I'll make a branch against this one.